### ISSUE TITLE
Updated version of libraries leads to decline in accuracy.

### ISSUE THREAD

**Issue by akshitsoni** [ageron/handson-ml#417]
I ran extra_capsnets.ipynb about 2 months ago 2/3/19 on google colab and the accuracies were coming as mentioned. But when I ran the same code today 5/5/19, the accuracy came down to (9.8200%). I believe, this might be due to an update in the version of libraries in Google colab.
Below is current version of libraries in google colab that i am using :-
TF: 1.13.1
Python: sys.version_info(major=3, minor=6, micro=7, releaselevel='final', serial=0)
NumPy: 1.16.3
The Answer: 42
GPU: False

Could you please let me know, which version update is causing this issue.

**Comment by ageron** [comment]
Hi @akshitsoni ,
Thanks a lot for your feedback. I'm finishing the 2nd edition of my book right now, so I'm not sure I'll have the time to investigate this issue right away, but as soon as I have a bit of time I'll try to see what version change caused this and why.

**Comment by ageron** [comment]
Yikes, I tried running it with TF 1.13.1 and my Jupyter kernel crashed!

```
OMP: Error #15: Initializing libiomp5.dylib, but found libiomp5.dylib already initialized.
OMP: Hint: This means that multiple copies of the OpenMP runtime have been linked into the program. That is dangerous, since it can degrade performance or cause incorrect results. The best thing to do is to ensure that only a single OpenMP runtime is linked into the process, e.g. by avoiding static linking of the OpenMP runtime in any library. As an unsafe, unsupported, undocumented workaround you can set the environment variable KMP_DUPLICATE_LIB_OK=TRUE to allow the program to continue to execute, but that may cause crashes or silently produce incorrect results. For more information, please see http://www.intel.com/software/products/support/.
```

**Comment by ageron** [comment]
Yes, I can confirm that after I did `conda uninstall tensorflow tensorflow-estimator tensorflow-base tensorboard` followed by `pip install tensorflow`, everything works normally now, training seems to be going well, the loss is down to 0.01358 after half of the first epoch.
Hope this helps.

**Comment by akshitsoni** [comment]
Hi @ageron ,
The training goes as expected for me also. The problem is with the validation accuracy as it goes down to 9% after the version update of tensorflow. It will be of great help if you can confirm whether you are getting validation accuracy as expected or not.  Thanks.

**Comment by ageron** [comment]
Hi @akshitsoni ,
Indeed, I also get lower validation accuracy, but not that low. I get 98.78% after the 1st epoch, but I couldn't let it run any longer. I'll run it again when I can.
Did you change anything in the notebook? Like the learning rate or batch size?
